pennylane/ftqc/lattice.py

```diff
@@ -0,0 +1,157 @@
+
+    This Lattice class, inspired by the design of :class:`~pennylane.spin.Lattice`, leverages `NetworkX` to represent the relationships within the lattice structure.
+
+        Args:
+            lattice_shape: Name of the lattice shape.
+            graph (nx.Graph): A NetworkX undirected graph object. If provided, `nodes` and `edges` are ignored.
+            nodes (List): Nodes to construct a graph object. Ignored if `graph` is provided.
+            edges (List): Edges to construct the graph. Ignored if `graph` is provided.
+        Raises:
+            ValueError: If neither `graph` nor both `nodes` and `edges` are provided.
+    """
+
+    # TODOs: To support braiding operations, Lattice should support nodes/edges addition/deletion.
+
+    def __init__(
+        self, lattice_shape: str, graph: nx.Graph = None, nodes: list = None, edges: list = None
+    ):
+        self._lattice_shape = lattice_shape
+        if graph is None:
+            if nodes is None and edges is None:
+                raise ValueError(
+                    "Neither a networkx Graph object nor nodes together with edges are provided."
+                )
+            self._graph = nx.Graph()
+            self._graph.add_nodes_from(nodes)
+            self._graph.add_edges_from(edges)
+        else:
+            self._graph = graph
+
+    @property
+    def get_lattice_shape(self) -> str:
+        r"""Returns the lattice shape name."""
+        return self._lattice_shape
+
+    def get_neighbors(self, node):
+        r"""Returns the neighbors of a given node in the lattice.
+
+        Args:
+            node: a target node label.
+        """
+        return self._graph.neighbors(node)
+
+    @property
+    def get_nodes(self):
+        r"""Returns all nodes in the lattice."""
+        return self._graph.nodes
+
+    @property
+    def get_edges(self):
+        r"""Returns all edges in the lattice."""
+        return self._graph.edges
+
+    @property
+    def get_graph(self) -> nx.Graph:
+        r"""Returns the underlying NetworkX graph object representing the lattice."""
+        return self._graph
+
+
+class LatticeShape(Enum):
+    """Enum to define valid set of lattice shape supported."""
+
+    chain = 1
+    square = 2
+    rectangle = 3
+    triangle = 4
+    honeycomb = 5
+    cubic = 6
+
+
+# map between lattice name and dimensions
+_LATTICE_DIM_MAP = {
+    "chain": 1,
+    "square": 2,
+    "rectangle": 2,
+    "cubic": 3,
+    "triangle": 2,
+    "honeycomb": 2,
+}
+
+# map between lattice name and networkx method name
+_LATTICE_GENERATOR_MAP = {
+    "chain": "grid_graph",
+    "square": "grid_graph",
+    "rectangle": "grid_graph",
+    "cubic": "grid_graph",
+    "triangle": "triangular_lattice_graph",
+    "honeycomb": "hexagonal_lattice_graph",
+}
+
+
+def generate_lattice(dims: list[int], lattice: str) -> Lattice:
+    r"""Generates a :class:`~pennylane.ftqc.Lattice` object with a given lattice dimensions and its shape name.
```

Comment:
```suggestion
    r"""Generates a :class:`~pennylane.ftqc.Lattice` object with a given lattice dimensions and its shape name.
```

Just wondering, is `dims` here a list of the extents ? As in, for cubic, it is a list of size 3 with each index giving the x/y/z lengths? I'm fine with `dims` as the name, but I'm wondering if we should clarify in the docstring. What do you think?